plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetPersister.java

```diff
@@ -895,7 +903,11 @@ private Throwable executeStatements(DBCSession session) {
                                 attributes[i] = statement.updateAttributes.get(i).getValue();
                             }
                             for (int i = 0; i < statement.keyAttributes.size(); i++) {
-                                attributes[statement.updateAttributes.size() + i] = statement.keyAttributes.get(i).getValue();
+                                if (DBUtils.isNullValue(statement.keyAttributes.get(i).getValue())) {
+                                    attributes[statement.updateAttributes.size() + i] = DBDNull.INSTANCE;
+                                } else {
+                                    attributes[statement.updateAttributes.size() + i] = statement.keyAttributes.get(i).getValue();
+                                }
```

Comment:
It would be nice to move this part to a separate method, because it's duplicated

Reply:
+1